### ISSUE TITLE
Show ChatGPT Responses Like Streams of Text (i.e., Word by Word)

### ISSUE THREAD

**Issue by AkibHossainOmi** [miroslavpejic85/mirotalksfu#188]
### Feature request

 Implement a feature to display ChatGPT responses as a stream of text, rendering output word by word or in small chunks in real time, instead of displaying the entire response at once. This would create a more dynamic and interactive user experience, similar to how text is streamed in ChatGPT.

### Pros

1. **Improved Perceived Speed**: 
   - Users can start reading the response as it’s being generated, reducing the feeling of waiting for the entire response.

2. **Enhanced User Experience**:
   - Simulates real-time conversational flow, making interactions feel more natural and engaging.

3. **Consistency with Industry Standards**:
   - Aligns with user interfaces of popular AI tools, such as OpenAI ChatGPT, that offer similar streaming response mechanisms.

4. **Better Feedback for Long Responses**:
   - For lengthy outputs, users can quickly identify the relevance of the response without waiting for it to finish.

5. **Psychological Engagement**:
   - Streaming responses tend to hold the user's attention better, as the gradual reveal keeps them focused.

**Comment by miroslavpejic85** [comment]
Implemented the feature to display ChatGPT responses as a `stream of text`, rendering output word by word or in small chunks in real time. This creates a dynamic and interactive experience, similar to how text is streamed in ChatGPT. Additionally, I have incorporated real-time `syntax highlighting` for any code blocks, ensuring they are properly formatted as they are displayed. I also `fixed the text-to-speech` functionality, allowing the content to be read aloud by clicking the specified button.

![mirotalk-sfu-chatgpt](https://github.com/user-attachments/assets/34d6bd05-e7be-4e27-acea-44d0381198d7)
